x-pack/plugins/monitoring/server/plugin.ts

```diff
@@ -0,0 +1,324 @@
+        }
+      });
+    } else if (!kibanaCollectionEnabled) {
+      log.info('Internal collection for Kibana monitoring is disabled per configuration.');
+    }
+
+    initInfraSource(config, legacyApi.infra);
+  }
+
+  start() {}
+
+  stop() {
+    if (this.cluster) {
+      this.cluster.close();
+    }
+  }
+
+  async getLegacyShim(
+    config: MonitoringConfig,
+    legacyConfig: any,
+    core: CoreSetup,
+    plugins: PluginsSetup,
+    licenseService: MonitoringLicenseService,
+    cluster: ICustomClusterClient
+  ) {
+    const router = core.http.createRouter();
+    const legacyConfigWrapper = () => ({
+      get: (_key: string): string => {
+        const key = _key.includes('monitoring.') ? _key.split('monitoring.')[1] : _key;
+        if (has(config, key)) {
+          return get(config, key);
+        }
+        if (has(legacyConfig, key)) {
+          return get(legacyConfig, key);
+        }
+
+        if (key === 'server.uuid') {
+          return core.uuid.getInstanceUuid();
+        }
+
+        throw new Error(`Unknown key '${_key}'`);
+      },
+    });
+    return {
+      config: legacyConfigWrapper,
+      log: this.log,
+      route: (options: any) => {
+        const method = options.method;
+        const handler = router.handleLegacyErrors(
+          async (
+            context: RequestHandlerContext,
+            req: KibanaRequest<any, any, any, any>,
+            res: KibanaResponseFactory
+          ) => {
+            const legacyRequest = {
+              ...req,
+              logger: this.log,
+              getLogger: this.getLogger,
+              payload: req.body,
+              getUiSettingsService: () => context.core.uiSettings.client,
+              server: {
+                config: legacyConfigWrapper,
+                newPlatform: {
+                  setup: {
+                    plugins,
+                  },
+                },
+                plugins: {
+                  monitoring: {
+                    info: licenseService,
+                  },
+                  elasticsearch: {
+                    getCluster: (name: string) => ({
+                      callWithRequest: async (_req: any, endpoint: string, params: any) => {
+                        const client =
+                          name === 'monitoring' ? cluster : core.elasticsearch.dataClient;
+                        return client?.asScoped(req).callAsCurrentUser(endpoint, params);
+                      },
+                    }),
+                  },
+                },
+              },
+            };
+            const result = await options.handler(legacyRequest);
+            return res.ok({ body: result });
+          }
+        );
+        const validate: any = get(options, 'config.validate', false);
+        if (validate && validate.payload) {
+          validate.body = validate.payload;
+        }
+        options.validate = validate;
+
+        if (method === 'POST') {
+          router.post(options, handler);
```

Comment:
if your legacy handler throws exceptions, you need to wrap it in https://github.com/elastic/kibana/blob/055c61110f8bbbdaea46ba5f28687d6692ccb97e/src/core/server/saved_objects/routes/delete.ts#L34
NP route handler doesn't support request rejecting via throwing exceptions anymore

Reply:
Yes, I'm doing this: https://github.com/elastic/kibana/pull/56675/files#diff-eef915a6b8dba3b904a62609b073163aR320